frontend/src/components/editor/errors/auto-fix.tsx

```diff
@@ -37,35 +40,65 @@ export const AutoFixButton = ({
   // multiple fixes.
   const firstFix = autoFixes[0];
 
+  const handleFix = () => {
+    const editorView =
+      store.get(notebookAtom).cellHandles[cellId].current?.editorView;
+    firstFix.onFix({
+      addCodeBelow: (code: string) => {
+        createNewCell({
+          cellId: cellId,
+          autoFocus: false,
+          before: false,
+          code: code,
+        });
+      },
+      editor: editorView,
+      cellId: cellId,
+      aiFix: {
+        setAiCompletionCell,
+        instantFix: instantAIFix,
+      },
+    });
+    // Focus the editor
+    editorView?.focus();
+  };
+
   return (
-    <Tooltip content={firstFix.description} align="start">
-      <Button
-        size="xs"
-        variant="outline"
-        className={cn("my-2 font-normal", className)}
-        onClick={() => {
-          const editorView =
-            store.get(notebookAtom).cellHandles[cellId].current?.editorView;
-          firstFix.onFix({
-            addCodeBelow: (code: string) => {
-              createNewCell({
-                cellId: cellId,
-                autoFocus: false,
-                before: false,
-                code: code,
-              });
-            },
-            editor: editorView,
-            cellId: cellId,
-            setAiCompletionCell,
-          });
-          // Focus the editor
-          editorView?.focus();
-        }}
-      >
-        <WrenchIcon className="h-3 w-3 mr-2" />
-        {firstFix.title}
-      </Button>
-    </Tooltip>
+    <div className={cn("flex gap-2 my-2 items-center", className)}>
+      <Tooltip content={firstFix.description} align="start">
+        <Button
+          size="xs"
+          variant="outline"
+          className="font-normal"
+          onClick={handleFix}
+        >
+          <WrenchIcon className="h-3 w-3 mr-2" />
+          {firstFix.title}
+        </Button>
+      </Tooltip>
+
+      {firstFix.fixType === "ai" && (
+        <div className="flex items-center gap-2">
+          <Switch
+            checked={instantAIFix}
+            onCheckedChange={() => setInstantAIFix(!instantAIFix)}
+            size="sm"
+            className="h-4 w-8"
+            title="Toggle instant AI fix mode"
+          />
+          <Tooltip
+            content={
+              instantAIFix ? "Instant fix enabled" : "Instant fix disabled"
+            }
+          >
+            {instantAIFix ? (
+              <ZapIcon className="h-3 w-3 text-amber-500" />
+            ) : (
+              <ZapOffIcon className="h-3 w-3 text-muted-foreground" />
```

Comment:
i would probably write this out instead of in a tooltip. there is still a good amount of realestate and might be confusing to hide it